src/power_grid_model/validation/validation.py

```diff
@@ -221,6 +221,20 @@ def validate_ids_exist(update_data: Dict[str, np.ndarray], input_data: SingleDat
     return list(chain(*errors))
 
 
+def process_power_sigma_and_p_q_sigma(data: SingleDataset, sensor: str, required_list: Dict[str, List[str]]) -> None:
+    """
+    Helper function to process the required list when both `p_sigma` and `q_sigma` exist
+    and valid but `power_sigma` is missing.
+    """
+    if sensor in data and isinstance(data[sensor], np.ndarray):
+        sensor_data = data[sensor]
+        if "p_sigma" in sensor_data.dtype.names and "q_sigma" in sensor_data.dtype.names:
+            p_sigma = sensor_data["p_sigma"]
+            q_sigma = sensor_data["q_sigma"]
+            if not (np.isnan(p_sigma).any() or np.isnan(q_sigma).any()):
```

Comment:
We cannot do this for all the elements. This has to be validated per element. So element 1 can have a valid `power_sigma`, and element 2 can have no valid `power_sigma`, but a valid `p_sigma` and `q_sigma`.

Reply:
The current code checks `p_sigma` and `q_sigma` for **all** the `sym_power_sensor`. We want to check this for each inidividual `sym_power_sensor`. So `sym_power_sensor_1` might have a valid `power_sigma` but no `p_sigma` or `q_sigma`. And `sym_power_sensor_1` might have a no `power_sigma` but valid `p_sigma` or `q_sigma`.

Reply:
I see what you mean. I'll take a look and make some changes accordingly (where necessary).

Reply:
Taken into account in the latest change.